src/truncated/normal.jl

```diff
@@ -118,6 +118,36 @@ function entropy(d::Truncated{<:Normal{<:Real},Continuous})
     0.5 * (log2π + 1.) + log(σ * z) + (aφa - bφb) / (2.0 * z)
 end
 
+function mgf(d::Truncated{<:Normal{<:Real},Continuous}, t::Real)
+    T = promote_type(partype(d), typeof(t))
```

Comment:
I guess
```suggestion
    T = float(promote_type(partype(d), typeof(t)))
```
would be more correct.